inspector/src/details/Property.ts

```diff
@@ -1,41 +1,84 @@
 module INSPECTOR {
-    
+
     /**
      * A property is a link between a data (string) and an object.
      */
     export class Property {
-        
+
         /** The property name */
-        private _property   : string;
+        private _property: string;
         /** The obj this property refers to */
-        private _obj        : any;
-        
-        constructor(prop:string, obj:any) {
+        private _obj: any;
+        /** The obj parent  */
+        private _parentObj: any;
+
+        constructor(prop: string, obj: any, parentObj?: PropertyLine) {
             this._property = prop;
             this._obj = obj;
+            this._parentObj = parentObj || null;
         }
-        
-        public get name() : string {
+
+        public get name(): string {
             return this._property;
         }
-        
-        public get value() : any {
+
+        public get value(): any {
             return this._obj[this._property];
         }
-        public set value(newValue:any) {
+        public set value(newValue: any) {
+            if (newValue != undefined && this._obj[this._property] != undefined) {
+                if (this._obj instanceof BABYLON.Scene) {
+                    (<BABYLON.Scene>this._obj).debugLayer.onGlobalPropertyChangeCallback({
+                        object: this._obj,
+                        property: this._property,
+                        value: newValue,
+                        initialValue: (<any>this._obj)[this._property]
+                    });
+                }
+                else {
+                    if(this._parentObj != null) {

```

Comment:
This is a confusing way of getting back to the debug layer. It would be cleaner if the observable was on the inspector and the inspector is passed into the property instead. The debug layer can then expose a separate observable that hooks to the inspector.

Reply:
I'm not sure I follow. #4464 just renames the property?